subprojects/kotlin-dsl-integ-tests/src/integTest/kotlin/org/gradle/kotlin/dsl/integration/KotlinDslAssignmentIntegrationTest.kt

```diff
@@ -0,0 +1,240 @@
+/*
+ * Copyright 2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.kotlin.dsl.integration
+
+import org.gradle.kotlin.dsl.assignment.internal.KotlinDslAssignment.ASSIGNMENT_SYSTEM_PROPERTY
+import org.gradle.kotlin.dsl.fixtures.AbstractKotlinIntegrationTest
+import org.gradle.test.fixtures.file.LeaksFileHandles
+import org.hamcrest.CoreMatchers.containsString
+import org.junit.Assert.assertEquals
+import org.junit.Test
+import java.io.File
+
+
+@LeaksFileHandles("Kotlin Compiler Daemon working directory")
+class KotlinDslAssignmentIntegrationTest : AbstractKotlinIntegrationTest() {
+
+    @Test
+    fun `can use assignment for properties in init scripts when assignment overload is enabled`() {
+        withAssignmentOverloadEnabled()
+        val initScript = withInitScriptWithAssignment()
+
+        // Expect
+        build("-I", initScript.canonicalPath).apply {
```

Comment:
❔ Do you need to use `canonicalPath` here? I'd expect it's okay to just pass the `initScript.absolutePath` or even just `initScript`.